pkg/policy/mutate/validate.go

```diff
@@ -81,6 +88,23 @@ func (m *Mutate) validateNestedForEach(tag string, j *v1.JSON) (string, error) {
 	return m.validateForEach(tag, nestedForeach)
 }
 
+func (m *Mutate) validatePatchStrategicMerge(psm apiextensions.JSON) error {
+	patchStrategicMergeJson, err := json.Marshal(psm)
```

Comment:
I did this because the `psm` variable should be of type `[]byte` in order to unmarshal it later and also `psm` variable wasn't getting converted to type `[]byte` using this statment `[]byte(psm)`. Below was the error which was being thrown.

```
cannot convert psm (variable of type apiextensions.JSON) to type []byte: need type assertion
```

So, in order to tackle this problem, I found a snippet of the code in the codebase which was tackling a similar problem. Below is the snippet for better illustration([pkg/validation/policy/validate.go#L610-L623](https://github.com/kyverno/kyverno/blob/main/pkg/validation/policy/validate.go#L610-L623)):
```
func validateElementInForEach(document apiextensions.JSON) error {
	jsonByte, err := json.Marshal(document)
	if err != nil {
		return err
	}

	var jsonInterface interface{}
	err = json.Unmarshal(jsonByte, &jsonInterface)
	if err != nil {
		return err
	}
	_, err = variables.ValidateElementInForEach(logging.GlobalLogger(), jsonInterface)
	return err
}
```